### PR TITLE
Refresh thumbnails when an image is re uploaded

### PR DESCRIPTION
**Pull request by fjvallarino**
## Description
This is part of https://github.com/Rise-Vision/rise-vision-apps/issues/1061. Design decisions have been discussed here (there is a change on how we decide when a thumbnail needs to be regenerated): https://docs.google.com/document/d/1a67JrnyuXlKwZwZnClBlpCH9HnFZvwm5xAjDsj9AMcs/edit#

## Motivation and Context
If an image is overwritten with a file of the same name but different content, the thumbnail is not updated correctly (the main file itself is). Re uploading files on Storage or Image Component results in stale thumbnails.

## How Has This Been Tested?
Several scenarios have been tested:
- Upload a new image from both Image Component or Storage. This image should use an Image Service thumbnail (can be checked by looking at `metadata.thumbnail` here: https://storage-dot-rvacore-test.appspot.com/_ah/api/storage/v0.02/files?companyId=87977ab8-38b6-47fb-ad5e-256b8cc4b46d&file=<file-path>)
- Re upload an image from both Image Component or Storage. The thumbnail should be custom. The url will start with `https://storage.googleapis.com/risemedialibrary-thumbnails-stage`)
- On Image Component, when uploading a file that already exists in the list, the thumbnail should be updated on all the instances and a new copy should _not_ be added.
- On Image Component, when selecting a file from Storage that already exists in the list, the thumbnail should be updated on all the instances and a new copy should be added.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
- Rollback of this changes depends on CCI. Going back to the previous build could be quick or not depending on CCI availability and number of e2e failures.
- The validation for production will be performed immediately after deployment

@alex-deaconu @andrecardoso @santiagonoguez @stulees @tejohnso please review. I mentioned you because you took part in the Design discussion, but it's not required all of you review the changes. Thanks!

### CODE REVIEW

File: web/scripts/template-editor/components/directives/dtv-component-image.js [hunk 1]
```diff
@@ -50,29 +50,42 @@ angular.module('risevision.template-editor.directives')
...
               file: filePath,
               exists: true,
-              'thumbnail-url': thumbnail
+              'thumbnail-url': _thumbnailFor(file)
             };
 
-            templateEditorUtils.addOrReplace(selectedImages, {
-              file: file.name
+            templateEditorUtils.addOrReplaceAll(selectedImages, {
+              file: filePath
```
Comment (santiagonoguez) [inline web/scripts/template-editor/components/directives/dtv-component-image.js]: @fjvallarino curious, using file.name was a bug before ?

Comment (fjvallarino) [reply web/scripts/template-editor/components/directives/dtv-component-image.js]: @santiagonoguez I think it was! It was missing the `companyId/` part of the path, so it was not finding the element in the list

**Comment by stulees** [comment]
@fjvallarino Changes look good to me also, however I see a number of `console.logs` added, just wondering if we require so many?

**Comment by fjvallarino** [comment]
@stulees some of them were already there (the ones in `dtv-upload`), but it's true we don't really need most of them. I can remove them if we think it's a better idea

**Comment by stulees** [comment]
@fjvallarino If you think it will help with validation on Production and ensure no issue, I'm good keeping them. Thanks!

**Pull request merged**